osu.Game/Screens/Play/PlayerSettings/BeatmapOffsetControl.cs

```diff
@@ -211,6 +220,27 @@ private void scoreChanged(ValueChangedEvent<ScoreInfo?> score)
                 return;
             }
 
+            unstableRate = hitEvents.CalculateUnstableRate();
+            const double unstablerate_threshold = 90;
+            const double exponential_factor = -0.0116;
+            // Parameters for the calculation that scales the offset with respect to UR.
+
+            if (autoAdjustBeatmapOffset.Value) //Enters this case if global setting is checked in Menu.
+            {
+                if (hitEvents.Count >= 10) //Make sure that UR and hit error average are available.
+                {
+                    Current.Value = unstableRate switch
+                    {
+                        >= unstablerate_threshold => lastPlayBeatmapOffset - (average * Math.Exp((double)(exponential_factor * (unstableRate - unstablerate_threshold)))),
+                        // A visual explanation of what is calculated here can be found in discussion #30521
+                        // A calculated offset which is achieved with low UR has higher impact than offset calculated with high UR.
+                        // "average" will be scaled by a value between 1.0 and 0.0
+                        < unstablerate_threshold => lastPlayBeatmapOffset - average,
+                        _ => Current.Value
+                    };
```

Comment:
Please replace this with plain `if` statements. We don't use switch expressions in our projects... yet.

Reply:
Done in commit [715a0f3](https://github.com/ppy/osu/pull/30586/commits/715a0f3fb76aa334ec79a40384ac0a5143091d8e)